pkg/blobstore/configuration/create_blob_access.go

```diff
@@ -496,6 +516,30 @@ func createDigestLocationMap(config *pb.LocalBlobAccessConfiguration, options *b
 		options.storageTypeName)
 }
 
+func createFileBackedDigestLocationMap(config *pb.LocalBlobAccessConfiguration, digestLocationMapConfig *pb.LocalBlobAccessConfiguration_FileBackedDigestLocationMap_, options *blobAccessCreationOptions, filename string) (local.DigestLocationMap, error) {
+	digestMapDirectory, err := filesystem.NewLocalDirectory(digestLocationMapConfig.FileBackedDigestLocationMap.Directory)
+	if err != nil {
+		return nil, err
+	}
+	defer digestMapDirectory.Close()
+	digestMapFile, err := digestMapDirectory.OpenReadWrite(filename, filesystem.CreateReuse(0644))
```

Comment:
I don't think `CreateReuse()` is safe just yet. If data is stored in memory, but the digest-location map is stored on disk (crazy I agree), it would cause us to reuse old digest-location map entries. For now I would suggest to use `CreateExcl()` and unlink the file up front.

Reply:
Alternative: do use `CreateReuse()`, but call `Truncate(0)` before truncating it to the desired size.

Reply:
Would this not completely empty the map? I realise this is unsafe with the InMemory backend, but a big benefit of this work is that it's necessary for persistent storage with the BlockDevice backend. I guess when that's added we'll have to add another case statement to check which backend is used and only empty the map for InMemory?

I'm realising just now that within the scope of this PR, this is also unsafe with BlockDevice backed storage as the Blocks don't get loaded on restart, so I'll definitely fix this up here in any case.

Reply:
"Yes" to the above! \o/

Reply:
So, I've added this now, however I've also added support for using a block device. As we can't truncate a block device, we will hit this problem again. I'm not well versed enough to know of some way to empty the array if it's backed by a block device. Any ideas?